### PR TITLE
[HttpFoundation] Fixed session migration with custom cookie lifetime

### LINKED ISSUE #28577: [HttpFoundation] PHP7.2 session migration error
Hello,

I upgrade my PHP version from PHP 5.6.36 to PHP 7.2, but I have a problem when I want to use the migrate function of Session component. 

My simple line of code :
`$this->request->getSession()->migrate(true, $newLifetime);`

The error returned :
`Warning: ini_set(): A session is active. You cannot change the session module's ini settings at this time `

I have already a session opened before by my login page, but I want to change the session life time on login.
My Symfony version 2.8.44
Thanks for your help

### PR DESCRIPTION
**Pull request by Guite**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #28577
| License       | MIT
| Doc PR        | 

This PR adds the fix proposed in https://github.com/symfony/symfony/issues/28577#issuecomment-578052397

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
How can we test this? There are some functional tests in this area, could be a fit?

**Comment by Guite** [comment]
@nicolas-grekas added a test case to `NativeSessionStorageTest`

**Comment by Guite** [comment]
FYI: appveyor failed because of `Symfony\Component\Process\Tests\ProcessTest::testStartIsNonBlocking`: `Failed asserting that 1.6472308635712 is less than 0.4.`

This seems not related, but flaky.

File: src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php [hunk 1]
```diff
@@ -214,7 +214,9 @@ public function regenerate($destroy = false, $lifetime = null)
         }
 
         if (null !== $lifetime) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php:216]: would it make sense to also change this line?
`if (null !== $lifetime && ini_get('session.cookie_lifetime') != $lifetime) {`
that's save saving+starting for nothing, isn't it?

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php (hunk 1 above)
Comment (Guite) [reply src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php]: indeed.
added the check in d303690442b73562b73d39e581762f1b022278bd

**Comment by nicolas-grekas** [comment]
Thank you @Guite.

**Pull request merged**